### PR TITLE
Read over Python 3.12's Logging Guide and see if there's anything else we could be leveraging/configuring to improve how logging is structured throughout our applications

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

This ticket consisted of reading the best practices and then changing our code to add 'excInfo=True' to most of our calls to current_app.logger.error().  By default, error() (unlike exception()) doesn't print a stack trace, but usually we want it to.  Then we needed to remove some of the raise Exception as e statements where we passed e as a parameter to the logger (since we would be getting the complete stack trace).

## Security Considerations

N/A

### CODE REVIEW

**Comment by ccostino** [comment]
Looks like this needs another merge from `main`, @terrazoon; could you also please update the PR description and details?  Thank you! 🙂

**Comment by ccostino** [comment]
D'oh, style check fail; also PR still needs a proper description! 🙂

**Review by ccostino — APPROVED** [review]
Awesome @terrazoon, thank you for this cleanup!  This will help improve logging going forward, and nice catch with several of the older debug statements or other things that should not be logged as well!

**Review by xlorepdarkhelm — CHANGES_REQUESTED** [review]
See my notes.

File: tests/app/test_cronitor.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def test_cronitor_does_nothing_if_name_not_recognised(notify_api, rmock, mocker)
         assert successful_task() == 1
 
     mock_logger.error.assert_called_with(
```
[the remark this answers is not in the document]
Comment (terrazoon) [reply tests/app/test_cronitor.py]: in this particular case, it is not because cronitor is explicitly doing an error message and not an exception.

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM

**Pull request merged**